packages/invoices/src/OutboundShipment/DetailView/Footer.tsx

```diff
@@ -110,10 +111,14 @@ export const Footer: FC<OutboundDetailFooterProps> = ({ draft, save }) => {
                   }}
                   variant="contained"
                   color="secondary"
-                  onClick={() => {
+                  onClick={async () => {
                     success('Saved invoice! 🥳 ')();
-                    draft.update?.('status', 'finalised');
-                    save(draft);
+                    await draft.update?.(
+                      'status',
+                      getNextOutboundStatus(draft?.status ?? '') ?? ''
+                    );
```

Comment:
as above, the `draft.update` isn't triggered until the end of the callback, so awaiting to fire the save afterward.

A little funny behaviour when saving the status, as we're optimistically updating the status, but the crumb is determined by the status dates (i.e. checks if there's a date to know the status which was easier to implement at the time..) so the button updates nicely however the crumb is slightly delayed as it waits for the response from the server with the new status date.

Could fix this by also optimistically updating the date for the status. However I wondered if we'd be putting a spinner up, anyway, in which case it wouldn't matter.

Could also be good to just have the button have a spinner

... Also debouncing probably

Reply:
actually quite like the status crumb updating just afterwards. seems neat, like something is happening. +1 for adding a spinner to the save & status buttons, and doing that would mean the status update makes more sense too